### ISSUE TITLE
systemd 234.11-9 : Randomly I get a "Failed unmounting /home"

### ISSUE THREAD

**Issue by Potomac** [systemd/systemd#6956]
Hello,

I use archlinux 64 bits,

since systemd 234.11-9 I get randomly an error message at reboot and shutdown :

systemd[1]: Unmounting /home...
systemd[1]: home.mount: Mount process exited, code=exited status=32
systemd[1]: Failed unmounting /home.

despite the error the reboot/shutdown works without problem ( and without additionnal delay ), I have only this error message every 4~5 reboots/shutdown,

it's the first time I see this problem, I use archlinux since 2013 and the unmounting process of /home has never failed, until the update of systemd for the version 234.11-9,

I am not sure if systemd is really responsible of this problem,  or if this error message is the consequence of an error from another process

thanks

**Comment by poettering** [comment]
anything relevant in the logs? anything relevant if you do this in debug mode? is your /home something special? luks? weird fs? LVM?

**Comment by Potomac** [comment]
Hello,

my /home is a hard-disk with a classical ext4 partition,

in journalctl I can see some errors related to X11 and Qt software, but I don't know it's important,

you can find as attached file the full journalctl log 
[systemd_mount_problem.txt](https://github.com/systemd/systemd/files/1349679/systemd_mount_problem.txt)

**Comment by marcosfrm** [comment]
I am also seeing this (Arch). Oddly, I do not have a separate /home. Just plain XFS / filesystem here.

```
[    3.009522] arch systemd-tmpfiles[175]: Reading config file "/usr/lib/tmpfiles.d/home.conf".
[    3.009870] arch systemd-tmpfiles[175]: Entry "/home" does not match any include prefix, skipping.
[    3.477292] arch systemd-tmpfiles[198]: Reading config file "/usr/lib/tmpfiles.d/home.conf".
[    3.511691] arch systemd-tmpfiles[198]: Running create action for entry Q /home
[    3.511821] arch systemd-tmpfiles[198]: Found existing directory "/home".
[    3.512219] arch systemd-tmpfiles[198]: Couldn't adjust quota for subvolume "/home" (unsupported fs or dir not a subvolume): Inappropriate ioctl for device
[    3.513933] arch systemd-tmpfiles[198]: "/home" has right mode 40755
[    3.514399] arch systemd-tmpfiles[198]: Running remove action for entry Q /home
[    9.541282] arch systemd[307]: home.mount: Failed to load configuration: No such file or directory
[    9.541665] arch systemd[307]: home-marcos.mount: Failed to load configuration: No such file or directory
[    9.575383] arch systemd[307]: home.mount: Collecting.
[    9.575739] arch systemd[307]: home-marcos.mount: Collecting.
[   12.000518] arch systemd[307]: home.mount: Failed to load configuration: No such file or directory
[   12.000941] arch systemd[307]: home-marcos.mount: Failed to load configuration: No such file or directory
[   12.002603] arch systemd[307]: home.mount: Collecting.
[   12.003040] arch systemd[307]: home-marcos.mount: Collecting.
[   12.139106] arch systemd[307]: home.mount: Failed to load configuration: No such file or directory
[   12.139545] arch systemd[307]: home-marcos.mount: Failed to load configuration: No such file or directory
[   12.145323] arch systemd[307]: home.mount: Collecting.
[   12.145590] arch systemd[307]: home-marcos.mount: Collecting.
[   12.325189] arch systemd[307]: home.mount: Failed to load configuration: No such file or directory
[   12.325629] arch systemd[307]: home-marcos.mount: Failed to load configuration: No such file or directory
[   12.327354] arch systemd[307]: home.mount: Collecting.
[   12.327663] arch systemd[307]: home-marcos.mount: Collecting.
[   12.482657] arch systemd[307]: home.mount: Failed to load configuration: No such file or directory
[   12.483106] arch systemd[307]: home-marcos.mount: Failed to load configuration: No such file or directory
[   12.484680] arch systemd[307]: home.mount: Collecting.
[   12.484999] arch systemd[307]: home-marcos.mount: Collecting.
[   12.654587] arch systemd[307]: home.mount: Failed to load configuration: No such file or directory
[   12.654946] arch systemd[307]: home-marcos.mount: Failed to load configuration: No such file or directory
[   12.656806] arch systemd[307]: home.mount: Collecting.
[   12.657170] arch systemd[307]: home-marcos.mount: Collecting.
[   13.184585] arch systemd[307]: home.mount: Failed to load configuration: No such file or directory
[   13.184959] arch systemd[307]: home-marcos.mount: Failed to load configuration: No such file or directory
[   13.187342] arch systemd[307]: home.mount: Collecting.
[   13.188709] arch systemd[307]: home-marcos.mount: Collecting.
[   67.979369] arch systemd[307]: home.mount: Failed to load configuration: No such file or directory
[   67.979758] arch systemd[307]: home-marcos.mount: Failed to load configuration: No such file or directory
[   67.991784] arch systemd[307]: home.mount: Collecting.
[   67.992162] arch systemd[307]: home-marcos.mount: Collecting.
```

[full-log.txt](https://github.com/systemd/systemd/files/1359054/full-log.txt)

**Comment by Mancooo** [comment]
Hi,

I am on Fedora 26 and running systemd.x86_64.233-6.fc26.

I also have the same problem as above. Shutdown continues normally and there is no delay, but sometimes I get this error:

okt 26 19:05:14 localhost.localdomain systemd[1]: Stopped Load/Save Random Seed.
okt 26 19:05:14 localhost.localdomain systemd[1]: Stopped Configure read-only root support.
okt 26 19:05:14 localhost.localdomain systemd[1]: home.mount: Mount process exited, code=exited status=32
okt 26 19:05:14 localhost.localdomain systemd[1]: Failed unmounting /home.
okt 26 19:05:14 localhost.localdomain systemd[1]: tmp.mount: Mount process exited, code=exited status=32
okt 26 19:05:14 localhost.localdomain systemd[1]: Failed unmounting Temporary Directory.
okt 26 19:05:14 localhost.localdomain systemd[1]: Stopped target Swap.
okt 26 19:05:14 localhost.localdomain systemd[1]: Deactivating swap /dev/disk/by-uuid/d7ae3d65-9eb0-4887-94cf-d7545b7c2e44...
okt 26 19:05:14 localhost.localdomain systemd[1]: Deactivating swap Swap Partition...
okt 26 19:05:14 localhost.localdomain systemd[1]: Stopped File System Check on /dev/disk/by-uuid/4a22b252-5923-4f14-bc6e-debae9f6b7df.
okt 26 19:05:14 localhost.localdomain audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-fsck@dev-disk-b
okt 26 19:05:14 localhost.localdomain systemd[1]: Unmounted /boot/efi.
okt 26 19:05:14 localhost.localdomain systemd[1]: Unmounting /boot...
okt 26 19:05:14 localhost.localdomain swapoff[1789]: swapoff: /dev/disk/by-uuid/d7ae3d65-9eb0-4887-94cf-d7545b7c2e44: swapoff failed: Invalid argument
okt 26 19:05:14 localhost.localdomain audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-fsck@dev-disk-b
okt 26 19:05:14 localhost.localdomain systemd[1]: Stopped File System Check on /dev/disk/by-uuid/D14B-581C.
okt 26 19:05:14 localhost.localdomain systemd[1]: dev-disk-by\x2duuid-d7ae3d65\x2d9eb0\x2d4887\x2d94cf\x2dd7545b7c2e44.swap: Swap process exited, code=exited status=255
okt 26 19:05:14 localhost.localdomain systemd[1]: Deactivated swap /dev/disk/by-uuid/d7ae3d65-9eb0-4887-94cf-d7545b7c2e44.
okt 26 19:05:14 localhost.localdomain systemd[1]: dev-disk-by\x2duuid-d7ae3d65\x2d9eb0\x2d4887\x2d94cf\x2dd7545b7c2e44.swap: Unit entered failed state.

**Comment by Mancooo** [comment]
Today I updated to Fedora 27 (beta) with systemd version **234-8.fc27** 

I get the same errors with failed unmounting /home and /temp

EDIT:

With debugging on, I get the following output in journalctl, hope it helps:

okt 27 22:55:18 localhost.localdomain systemd[1]: Stopped target Local File Systems.
okt 27 22:55:18 localhost.localdomain systemd[1]: tmp.mount: About to execute: /usr/bin/umount /tmp -c
okt 27 22:55:18 localhost.localdomain systemd[1]: tmp.mount: Forked /usr/bin/umount as 1655
okt 27 22:55:18 localhost.localdomain systemd[1]: tmp.mount: Changed mounted -> unmounting
okt 27 22:55:18 localhost.localdomain systemd[1]: Unmounting Temporary Directory (/tmp)...
okt 27 22:55:18 localhost.localdomain systemd[1]: home.mount: About to execute: /usr/bin/umount /home -c
okt 27 22:55:18 localhost.localdomain systemd[1]: tmp.mount: About to execute: /usr/bin/umount /tmp -c
okt 27 22:55:18 localhost.localdomain systemd[1]: tmp.mount: Forked /usr/bin/umount as 1655
okt 27 22:55:18 localhost.localdomain systemd[1]: tmp.mount: Changed mounted -> unmounting
okt 27 22:55:18 localhost.localdomain systemd[1]: Unmounting Temporary Directory (/tmp)...
okt 27 22:55:18 localhost.localdomain systemd[1]: home.mount: About to execute: /usr/bin/umount /home -c
okt 27 22:55:18 localhost.localdomain systemd[1]: home.mount: Forked /usr/bin/umount as 1656
okt 27 22:55:18 localhost.localdomain systemd[1]: home.mount: Changed mounted -> unmounting
okt 27 22:55:18 localhost.localdomain systemd[1]: Unmounting /home...
okt 27 22:55:18 localhost.localdomain systemd[1655]: tmp.mount: Executing: /usr/bin/umount /tmp -c
okt 27 22:55:18 localhost.localdomain systemd[1]: boot-efi.mount: About to execute: /usr/bin/umount /boot/efi -c
okt 27 22:55:18 localhost.localdomain systemd[1]: boot-efi.mount: Forked /usr/bin/umount as 1657
okt 27 22:55:18 localhost.localdomain systemd[1]: boot-efi.mount: Changed mounted -> unmounting
okt 27 22:55:18 localhost.localdomain systemd[1]: Unmounting /boot/efi...
okt 27 22:55:18 localhost.localdomain systemd[1656]: home.mount: Executing: /usr/bin/umount /home -c
okt 27 22:55:18 localhost.localdomain systemd[1]: fedora-readonly.service: Changed exited -> dead
okt 27 22:55:18 localhost.localdomain systemd[1]: fedora-readonly.service: Job fedora-readonly.service/stop finished, result=done
okt 27 22:55:18 localhost.localdomain systemd[1]: Stopped Configure read-only root support.
okt 27 22:55:18 localhost.localdomain systemd[1657]: boot-efi.mount: Executing: /usr/bin/umount /boot/efi -c
okt 27 22:55:18 localhost.localdomain systemd[1]: libmount event [rescan: yes]
okt 27 22:55:18 localhost.localdomain systemd[1]: Received SIGCHLD from PID 1656 (umount).
okt 27 22:55:18 localhost.localdomain systemd[1]: Child 1655 (umount) died (code=exited, status=32/n/a)
okt 27 22:55:18 localhost.localdomain systemd[1]: tmp.mount: Child 1655 belongs to tmp.mount
okt 27 22:55:18 localhost.localdomain systemd[1]: tmp.mount: Mount process exited, code=exited status=32
okt 27 22:55:18 localhost.localdomain systemd[1]: tmp.mount: Changed unmounting -> mounted
okt 27 22:55:18 localhost.localdomain systemd[1]: tmp.mount: Job tmp.mount/stop finished, result=failed
okt 27 22:55:18 localhost.localdomain systemd[1]: Failed unmounting Temporary Directory (/tmp).
okt 27 22:55:18 localhost.localdomain systemd[1]: Child 1656 (umount) died (code=exited, status=32/n/a)
okt 27 22:55:18 localhost.localdomain systemd[1]: home.mount: Child 1656 belongs to home.mount
okt 27 22:55:18 localhost.localdomain systemd[1]: home.mount: Mount process exited, code=exited status=32
okt 27 22:55:18 localhost.localdomain systemd[1]: home.mount: Changed unmounting -> mounted
okt 27 22:55:18 localhost.localdomain systemd[1]: home.mount: Job home.mount/stop finished, result=failed
okt 27 22:55:18 localhost.localdomain systemd[1]: Failed unmounting /home.
okt 27 22:55:18 localhost.localdomain systemd[1]: systemd-fsck@dev-disk-by\x2duuid-4a22b252\x2d5923\x2d4f14\x2dbc6e\x2ddebae9f6b7df.service: Changed exited -> dead
okt 27 22:55:18 localhost.localdomain systemd[1]: systemd-fsck@dev-disk-by\x2duuid-4a22b252\x2d5923\x2d4f14\x2dbc6e\x2ddebae9f6b7df.service: Job systemd-fsck@dev-disk-by\
okt 27 22:55:18 localhost.localdomain systemd[1]: Stopped File System Check on /dev/disk/by-uuid/4a22b252-5923-4f14-bc6e-debae9f6b7df.
okt 27 22:55:18 localhost.localdomain systemd[1]: swap.target changed active -> dead
okt 27 22:55:18 localhost.localdomain systemd[1]: swap.target: Job swap.target/stop finished, result=done
okt 27 22:55:18 localhost.localdomain systemd[1]: Stopped target Swap.

**Comment by Mancooo** [comment]
After installing (not upgrading) Fedora 27 on my main desktop, I get the same error about failed unmounting /home.

On my laptop with Fedora 27 I also still get these messages, now on every shutdown. I recently installed Fedora 27 on my father's laptop, he also has the same error.

All three systems are on systemd.234-9.fc27

**Comment by Mancooo** [comment]
This issue still exists on Fedora 28 (systemd-238-5.fc28). I have tried different things for hours and still can't find a solution.

I run Gnome 3.28 without gdm (so startx via multi-user.target). Everything works fine. However whenever play audio and after that log out to the console again. I see with lsof /home that there are a small number of open files. The only interesting things are the files related to Pulseaudio in my ~/.config/pulse.

So I figure it has something to do with Pulseaudio in my case. When I quit from X11 and do pulseaudio -k and then reboot or shutdown, than no error concerning failed unmounting occurs.

I have spend many hours trying to debug this problem. I have the issue on both of my laptops and my desktop (1 running Fedora 28 and 2 running Fedora 27).

What can I do to help?

**Comment by boucman** [comment]
@Mancooo this looks a lot like a pulseaudio bug... pulseaudio needs to be stoped before /home is unmounted, which is something that needs to be specified by the RequiresMountFor somewhere... probably in whatever service contains the pulseaudio daemon... That is either your specific startx service or a pulseaudio-provided service... both of which arn't maintained here.

**Comment by Mancooo** [comment]
@boucman Thank you for your comment.

I think you are right. It is something with Pulseaudio. But the thing is I don't use any service. I have a minimal installation. It works great, I just have this problem for a long time and I am afraid of data corruption.

I start Gnome just with startx and Pulseaudio starts automatically. I looked but Fedora does not have a pulseaudio.service or pulseaudio.socket. Out of frustration I just wiped one of my laptops and will start over again. I am 100% sure the problem will arise again after reinstallation, in that case I will submit a bug at the Pulseaudio bugzilla.

Thanks again.
